src/main/java/org/terasology/taskSystem/TaskManagementSystem.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.taskSystem;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.engine.Time;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.holdingSystem.components.HoldingComponent;
+import org.terasology.logic.behavior.core.Actor;
+import org.terasology.logic.selection.ApplyBlockSelectionEvent;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.minion.move.MinionMoveComponent;
+import org.terasology.registry.In;
+import org.terasology.registry.Share;
+import org.terasology.taskSystem.components.TaskComponent;
+import org.terasology.world.selection.BlockSelectionComponent;
+
+import java.util.List;
+
+@Share(TaskManagementSystem.class)
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class TaskManagementSystem extends BaseComponentSystem {
+    private static final Logger logger = LoggerFactory.getLogger(TaskManagementSystem.class);
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private Time timer;
+
+    private HoldingComponent oreonHolding;
+
+    public void setOreonHolding(HoldingComponent holding) {
+        this.oreonHolding = holding;
+    }
+
+    public boolean getTaskForOreon(Actor oreon) {
+        List<EntityRef> availableTasks = oreonHolding.availableTasks;
```

Comment:
At the moment, a  Deque<E> would be a better data structure than a List for storing tasks.
It allows you to see if there are any tasks waiting, pull off the next one as well as add new tasks.
It will also automatically sort by task addition time for you.